venvctrl/venv/base.py

```diff
@@ -167,15 +172,28 @@ def shebang(self, new_shebang):
             ValueError: If the file has no shebang to modify.
             ValueError: If the new shebang is invalid.
         """
+
         if not self.shebang:
 
             raise ValueError('Cannot modify a shebang if it does not exist.')
 
-        if not new_shebang.startswith('#!'):
+        if new_shebang is None:
+
+            raise ValueError('New shebang cannot be None.')
+
+        new_shebang = new_shebang.split(os.linesep)
+
+        if len(self.shebang) != len(new_shebang):
```

Comment:
There's a test failure here because it's comparing the length of the shebang string against the length of the new_shebang list.

Reply:
Oops, should be resolved now.

Reply:
Still failing, investigating

Reply:
Alright it's really resolved now 😄